### PR TITLE
[TS SDK] Explore http2

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description
Explore http2 using [fetch-h2 package](https://github.com/grantila/fetch-h2). 

Notes:
* We need a completely new client. Currently we use `axios` but it doesnt support http2.
* sleep for 15 seconds when funding accounts to not spend time implementing `waitFoTransaction` with this new client.
* when trying to send 1000 read requests the client explodes with `Maximum call stack size exceeded` - seems like a known issue https://github.com/grantila/fetch-h2/issues/93. Also checked the same process on the low-level http2 node module (what fetch-h2 is based on) and getting a `Stream closed with error code NGHTTP2_REFUSED_STREAM` error although they claim they fixed it https://github.com/nodejs/node/issues/27416
* some account created and used during tests
https://explorer.aptoslabs.com/account/0x2230fa1a57bf2a1a1e22dcdcd9ef5ce99a0330d5bdb470fc6bc7b4d923d07175
https://explorer.aptoslabs.com/account/0xaad9a573b24d805fe9faa7b29e4c74ce50fe0a48346ce43c35152841faf41abb
https://explorer.aptoslabs.com/account/0x8e61b5b6d0cf993f37050c1c7a77dde01c0c845231db1521f57e66942a537e1b
https://explorer.aptoslabs.com/account/0x89f548868fae1ca803776c85d37b29cfa6b5ab722d5364f02cf3f30ee6e6edfe
All end up with sequence number = 99 - which makes sense since we submitted 100 txns per account.
```
starting...
accounts created
{ ms: 34, s: 0, m: 0, h: 0, d: 0 }
accounts funded
{ ms: 305, s: 15, m: 0, h: 0, d: 0 }
accounts checked
{ ms: 708, s: 16, m: 0, h: 0, d: 0 }
accounts initialized
{ ms: 784, s: 16, m: 0, h: 0, d: 0 }
transactions submitted
{ ms: 19, s: 38, m: 0, h: 0, d: 0 }
transactions commited
{ ms: 675, s: 39, m: 0, h: 0, d: 0 }
```

Thank you David for code inspiration.

### CODE REVIEW

File: ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+  }
+
+  await Promise.all(waitFor);
+  console.log("transactions commited");
+  console.log(timer.time());
+
+  exit(0);
+}
+
+async function transafer(sender: AptosAccount, recipient: string, sequenceNumber: bigint, amount: number) {
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts:101]: transfer or tran safer? 😂

File: ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  const token = new TxnBuilderTypes.TypeTagStruct(TxnBuilderTypes.StructTag.fromString("0x1::aptos_coin::AptosCoin"));
+
+  const entryFunctionPayload = new TxnBuilderTypes.TransactionPayloadEntryFunction(
+    TxnBuilderTypes.EntryFunction.natural(
+      "0x1::coin",
+      "transfer",
```
Comment (gregnazario) [inline ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts:107]: fyi, for future ease of use: `0x1::aptos_account::transfer_coins` will create the account / register the coin if it doesn't exist

File: ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+  console.log("starting...");
+  timer.start();
+  // create accounts
+  const accounts: AptosAccount[] = [];
+  const recipients: AptosAccount[] = [];
+  for (let i = 0; i < accountsCount; i++) {
+    accounts.push(new AptosAccount());
+    recipients.push(new AptosAccount());
+  }
+  console.log(`${accounts.length * 2} accounts created`);
```
Comment (0xjinn) [inline ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts:29]:
```suggestion
  console.log(`${accounts.length + recipients.length} accounts created`);
```

nit: Number still the same, but i felt like its more read-able this way :D

File: ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+  for (let i = 0; i < recipients.length; i++) {
+    funds.push(`${FAUCET_URL}/mint?address=${HexString.ensure(recipients[i].address()).noPrefix()}&amount=${0}`);
+  }
+  // send requests
+  const responses = await Promise.all(
+    funds.map((fund) =>
+      fetch(fund, {
+        method: "POST",
+        headers: {
+          Authorization: `Bearer ejfklsdfj7vr4388fhhfh3f78hf345345hf00da0`,
```
Comment (0xjinn) [inline ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts:47]: Curious, what is this bearer token here?

File: ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    BigInt(100),
+    // Expiration timestamp. Transaction is discarded if it is not executed within 20 seconds from now.
+    BigInt(Math.floor(Date.now() / 1000) + 20),
+    new TxnBuilderTypes.ChainId(2),
+  );
+
+  const bcsTxn = AptosClient.generateBCSTransaction(sender, rawTransaction);
+  return bcsTxn;
+  // const txn = await submitTransaction(bcsTxn);
+  // return txn.hash;
```
Comment (0xjinn) [inline ecosystem/typescript/sdk/examples/typescript/http2-fetch-h2/index.ts:165]: nit: Forgot to remove these 2 lines?

**Review by 0xjinn — APPROVED** [review]

**Pull request closed without merging**